CHANGELOG.md

```diff
@@ -1,54 +1,90 @@
+  * renamed ns `spec-tools.conform` into `spec-tools.transformer`, covering both encoding & decoding of values
+  * `st/type-transformer`, supporting both `:type`-level transformations & `Spec` level transformations via new Spec keys with `encode` and `decode` namespaces + transformer name as name.
+  * `st/encode`, `st/decode`, `st/explain`, `st/explain-data`, `st/conform` and `st/conform!` take the transformer instance an optional third argument
+     * `st/json-transformer` and `st/string-transformer` are shipped out-of-the-box
+
+### Transformer
+
+```clj
+(defprotocol Transformer
+  (-transformer-name [this])
+  (-encoder [this spec value])
+  (-decoder [this spec value]))
+```
+
+### Type-driven transformations
+
+* use `:type` information from Specs (mostly resolved automatically)
 
 ```clj
-(require '[clojure.spec.alpha :as s])
 (require '[spec-tools.core :as st])
 
+(as-> "2014-02-18T18:25:37Z" $
+      (st/decode inst? $))
+; :clojure.spec.alpha/invalid
+
+;; decode using string-transformer
+(as-> "2014-02-18T18:25:37Z" $
+      (st/decode inst? $ st/string-transformer))
+; #inst"2014-02-18T18:25:37.000-00:00"
+
+(as-> "2014-02-18T18:25:37Z" $
+      (st/decode inst? $ st/string-transformer)
+      (st/encode inst? $))
+; :clojure.spec.alpha/invalid
+
+;; encode using string-transformer
+(as-> "2014-02-18T18:25:37Z" $
+      (st/decode inst? $ st/string-transformer)
+      (st/encode inst? $ st/string-transformer))
+; "2014-02-18T18:25:37.000+0000"
+```
+
+### Spec-driven transformations
+
+* use `:encode/*` and `:decode/*` keys from Spec instances
+
+```clj
+(require '[clojure.spec.alpha :as s])
+(require '[clojure.string :as str])
+
 (s/def ::spec
   (st/spec
     {:spec #(and (simple-keyword? %) (-> % name str/lower-case keyword (= %)))
      :description "a lowercase simple keyword, encoded in uppercase in string-mode"
      :decode/string #(-> %2 name str/lower-case keyword)
      :encode/string #(-> %2 name str/upper-case)}))
 
-; decode also validates
-(st/decode ::spec "kikka")
-; => :clojure.spec.alpha/invalid
-
-(st/decode ::spec "kikka" st/string-conforming)
-; => :kikka
-
-; encode fails if no encoder present
-(st/encode ::spec "kikka")
-; => :clojure.spec.alpha/invalid
-
-; encode doesn't validate!
-(st/encode ::spec "kikka" st/string-conforming)
-; => "KIKKA"
-
-; not real bijections (https://en.wikipedia.org/wiki/Bijection)
-(as-> "KikKa" $
-      (doto $ prn)
-      (st/encode ::spec $ st/string-conforming)
-      (doto $ prn)
-      (st/decode ::spec $ st/string-conforming)
-      (doto $ prn)
-      (st/encode ::spec $ st/string-conforming)
-      (prn $))
-; "KikKa"
-; "KIKKA"
+(st/decode ::spec :kikka)
 ; :kikka
+
+(as-> "KiKka" $
+      (st/decode ::spec $))
+; :clojure.spec.alpha/invalid
+
+(as-> "KiKka" $
+      (st/decode ::spec $ st/string-transformer))
+; :kikka
+
+(as-> "KiKka" $
+      (st/decode ::spec $ st/string-transformer)
+      (st/encode ::spec $))
+; :clojure.spec.alpha/invalid
+
+(as-> "KiKka" $
```

Comment:
Should we keep here the comment about bijections? (first time I read it I thought it was a nice disclaimer)